django_saml2_auth/views.py

```diff
@@ -249,5 +275,58 @@ def signin(r):
 
 
 def signout(r):
+    """
+    Initiates saml2 logout.
+
+    :param r: HTTP request
+    """
+    saml_client = _get_saml_client(get_current_domain(r))
+    subject_id = _get_subject_id(r.session)
+    idp_entity_id = None
+    try:
+        idp_entity_id = settings.SAML2_AUTH['SAML_CLIENT_SETTINGS']['service']['sp']['idp']
+    except KeyError:
+        return HttpResponseServerError('Idp not defined')
+
+    response = saml_client.do_logout(subject_id, [idp_entity_id], "", None)
+    return handle_logout_response(response)
+
+
+def handle_logout_response(response):
+    """
+    Handles saml2 logout response.
+
+    :param response: Saml2 logout response
+    """
+    if len(response) > 1:
+        # Currently only one source is supported
+        return HttpResponseServerError('Logout from several sources not supported')
+    for entityid, logout_info in response.items():
+        if isinstance(logout_info, tuple):
+            # logout_info is a tuple containing header information and a HTML message.
+            binding, http_info = logout_info
+            if binding == BINDING_HTTP_POST:
+                # Display content defined in logout response
+                body = ''.join(http_info['data'])
+                return HttpResponse(body)
+            elif binding == BINDING_HTTP_REDIRECT:
+                # Redirect to address defined in logout response
+                return HttpResponseRedirect(_get_location(http_info))
+            else:
+                # Unknown binding
+                return HttpResponseServerError('Logout binding not supported')
+        else:
+            # Soap logout does not return tupple and does not call callback so we logout locally here
+            return finish_signout(r, logout_info)
```

Comment:
What is "Soap logout"?

You should also test all these branches. This at least is using a parameter that is undefined.

Reply:
Removed this as we cant test it.